ui/src/message_popup/author/questions_page.jsx

```diff
@@ -29,12 +30,18 @@ export default React.createClass({
 
   getInitialState(){
     return ({
-      questions: allQuestions,
+      allDatabaseQuestions: [],
+      currentQuestions: [],
+      archivedQuestions: [],
```

Comment:
Ah yeah that totally makes sense, and yep the simplification comes from deferring computation of filtering the list to render, hopefully that helps simplify :)